server/utils/strapi4Utils.ts

```diff
@@ -36,8 +37,6 @@ export function unwrapSingleResponse<T extends Record<string, unknown>>(response
   return unwrapAttributes<T>(data)
 }
 
-export function unwrapListResponse<T extends Record<string, unknown>>(
-  response: ListResponse<T>,
-): Array<T & { id: number }> {
-  return response.data.map(({ id, attributes }) => ({ ...(unwrapAttributes(attributes) as T), id }))
+export function unwrapListResponse<T extends Record<string, unknown>>(response: ListResponse<T>): Unwrapped<T>[] {
+  return response.data.map(({ id, attributes }) => ({ ...unwrapAttributes<T>(attributes), id }))
```

Comment:
The `unwrapAttributes` being able to understand arrays means we could replace this with:

`unwrapAttributes<T>(response.data) as ...loads of dodgy casting`

but think your `response.data.map` is more easy to understand!